File: 1 Python/solutions/lab01.py
# Unit Converter

# Version 1

#feet = input("What is the distance in feet? ")
#feet = int(feet)

#metersfeet = feet * .3048
#metersfeet = float(metersfeet)

#print(f"{feet} ft is {metersfeet} m")

# Version 2 & 3

# distance = input("What is the distance? ")
# distance = float(distance)

# unit = input("What are the units? ")

# ft = .3048
# mi = 1609.34
# m = 1.0
# km = 1000.0
# yd = .9144
# inch = .0254

# if unit == "feet":
#     meters = distance * ft
#     print(f"{distance} ft is {meters} m")
# elif unit == "miles":
#     meters = distance * mi
#     print(f"{distance} mi is {meters} m")
# elif unit == "meters":
#     meters = distance * m
#     print(f"{distance} m is {meters} m")
# elif unit == "kilometers":
#     meters = distance * km
#     print(f"{distance} km is {meters} m")
# elif unit == "yards":
#     meters = distance * yd
#     print(f"{distance} yd is {meters} m")
# elif unit == "inches":
#     meters = distance * inch
#     print(f"{distance} in is {meters} m")

# Version 4

distance = input("What is the distance? ")
distance = float(distance)

inputunit = input("What are the input units? ")

output = input("What are the output units? ")


if output == "feet":
    output = .3048
    outputunit = "ft"
elif output == "miles":
    output = 1609.34
    outputunit = "mi"
elif output == "meters":
    output = 1.0
    outputunit = "m"
elif output == "kilometers":
    output = 1000.0
    outputunit = "km"


ft = .3048
mi = 1609.34
m = 1.0
km = 1000.0


if inputunit == "feet":
    meters = distance * ft
    conversion = meters / output
    conversion = float(conversion)
    print(f"{distance} ft is {conversion} {outputunit}")
elif inputunit == "miles":
    meters = distance * mi
    conversion = meters / output
    conversion = float(conversion)
    print(f"{distance} mi is {conversion} {outputunit}")
elif inputunit == "meters":
    meters = distance * m
    conversion = meters / output
    conversion = float(conversion)
    print(f"{distance} m is {conversion} {outputunit}")
elif inputunit == "kilometers":
    meters = distance * km
    conversion = meters / output
    conversion = float(conversion)
    print(f"{distance} km is {conversion} {outputunit}")

File: code/sarde/javascript/lab01_A/lab01.js
// Lab 1: Unit Converter
/*
Version 1
-Ask the user for the number of feet
-print equivalent distance in meters
- 1 ft is 0.3048 m.
-output = meters = multiply input distance by 0.3048

let units = {
    feet: 0.3048
}
console.log(units)

//ask user for number of feet
let inputFeet = prompt('What is the distance in feet?')
// console.log(typeof input_feet)
feet = parseInt(input_feet)
// console.log(typeof feet)
let total = feet * units['feet']
// console.log(Math.round(total, 4))
alert(`${feet} ft is ${Math.round(total, 4)} m`)
*/
//-------------------------------------------------------------------------------------------------------------------------//

/*
Version 2
-Allow the user to also enter the units
-depending on the unit, convert distance-meters


let units = {
    ft: 0.3048,
    mi: 1609.34,
    m: 1,
    km: 1000
}
console.log(units)

// ask the user to enter the distance
let inputDistance = prompt('What is the distance?')
// ask the user to enter the units
let inputUnit = prompt('What is the unit(ft, mi, m, km)?')
distance = parseInt(inputDistance)//convert to a number
let total = distance * units[inputUnit]
// console.log(Math.round(total))
alert(`${distance} ${inputUnit} is ${total.toFixed(2)} m`)
*/
//------------------------------------------------------------------------------------------------------------------//

/*
Version 3
-Add support for yards, and inches
-1 yard is 0.9144 m
-1 inch is 0.0254 m


let units = {
    ft: 0.3048,
    mi: 1609.34,
    m: 1,
    km: 1000,
    yd: 0.9144,
    in: 0.0254
}
console.log(units)

//ask the user to enter the distance
let inputDistance = prompt('What is the distance?')
//ask the user to enter the unit
let inputUnit = prompt('What is the unit(ft, mi, m, km, yd, in)?')
//convert inputDistance to a Number
let distance = parseInt(inputDistance)
let total = distance * units[inputUnit]
// console.log(total.toFixed(2))
alert(`${distance} ${inputUnit} is ${total.toFixed(2)} m`)
*/
//--------------------------------------------------------------------------------------------------------------------//

/*
Version 4
-Ask the user for distance
-Ask the user for input_units
-Ask the user for output_units
-convert any unit to meters
-covert distance in meters to any other unit
Note:convert from meters by dividing distance(in meters)
        -convert from input_units to meters, then
        convert from meters to output_units
*/
let units = {
    'ft': 0.3048,
    'mi': 1609.34,
    'm': 1,
    'km': 1000,
    'yd': 0.9144,
    'in': 0.0254
}
console.log(units)

let inputDistance = prompt('What is the distance?')
let inputUnit = prompt('What is the unit(ft, mi, m, km, yd, in)?')
let outputUnit = prompt('What is the output unit(ft, mi, m, km, yd, in)?')
// convert any input_unit to meters
let convertedInputUnit = units[inputUnit]

//convert from meters to output_unit
let convertedOutputUnit = units[outputUnit]
let total = inputDistance * convertedInputUnit
// console.log(total)

alert(`${total / convertedOutputUnit}`)